lib/rubocop/cop/style/redundant_percent_q.rb

```diff
@@ -53,13 +53,17 @@ def check(node)
           return if interpolated_quotes?(node) || allowed_percent_q?(node)
 
           add_offense(node) do |corrector|
-            delimiter = /^%Q[^"]+$|'/.match?(node.source) ? QUOTE : SINGLE_QUOTE
+            delimiter = delimiter_for(node.source)
```

Comment:
Can you update to the following instead of creating the new `delimiter_for` method:

```suggestion
            delimiter = /\A%Q[^"]+\z|'/.match?(node.source) ? QUOTE : SINGLE_QUOTE
```